api/lib/application/assessments/assessment-controller.js

```diff
@@ -92,8 +92,9 @@ module.exports = {
 
     await DomainTransaction.execute(async (domainTransaction) => {
       const assessmentCompletedEvent = await usecases.completeAssessment({ domainTransaction, assessmentId });
-      await events.handleCertificationScoring({ assessmentCompletedEvent });
+      const certificationScoringEvent = await events.handleCertificationScoring({ domainTransaction, assessmentCompletedEvent });
       await events.handleBadgeAcquisition({ domainTransaction, assessmentCompletedEvent });
+      await events.handleCertificationAcquisitionForPartner({ domainTransaction, certificationScoringEvent });
```

Comment:
```suggestion
      if (certificationScoringEvent) {
        await events.handleCertificationAcquisitionForPartner({ domainTransaction, certificationScoringEvent });
      }
```
(en lien avec le commentaire sur le retour de `handleCertificationScoring`)

Reply:
J'aurais tendance à dire qu'il vaut mieux que ce soit handleCertificationAcquisitionForPartner qui gere cette condition. C'est lui qui sait s'il doit être executé (ce sera si c'est une certification).

Reply:
Ah oui, intéressant. Pour moi ici on est en train d’évoluer vers un pattern Publish-Subscribe, où le contrôleur joue temporairement le rôle du broker / event bus : donc j’aime bien la proposition du `if` dans le contrôleur, qui permet de ne pas notifier les handlers si le message ne les concerne pas (comme le ferait un broker, qui ne notifierait les handlers abonnées à l’événement `certificationScoringEvent` que quand un message de ce type se produit).

Reply:
Oui c'était à ça que je pensais. C'est clair pour moi que l'écriture actuelle où le contrôleur sait quels événements attendre et à qui les envoyer est provisoire.

Reply:
OK, j'ai rajouté le if, je ne trouve pas ca propre, parce qu'effctivement le controlleur fait le boulot du broker mais dans l'immediat, c'est plus logique.

Reply:
D’ailleurs pour moi après cette PR on aura assez d’information pour commencer à expérimenter et proposer un mécanisme de pub sub dans une autre PR. J’ai un peu testé l’`EventEmitter` de Node.js, on peut l’utiliser comme broker / dispatcher par exemple. J’aurais le réflexe de l’encapsuler dans une classe à nous pour pouvoir faire évoluer son API indépendamment de celle d’`EventEmitter` et pour pouvoir le substituer facilement dans les tests. On peut aussi faire une implémentation à nous ou trouver une lib qui fait ça, à débattre, je n’ai pas poussé la réflexion très loin.